test/tests/functional/pbs_node_rampdown.py

```diff
@@ -6362,3 +6415,774 @@ def test_release_nodes_shared_server_restart_immed(self):
+            for key in self.server.pu.processes:
+                if ("fib" in key):
+                    process = len(self.server.pu.processes[key])
+                    print "length of the process is %d " % (process,)
+
+        self.assertEqual(process, 3)
+
+        # Release node2 from job1 only
+        cmd = [self.pbs_release_nodes_cmd, '-j', jid1, self.hostB]
+        ret = self.server.du.run_cmd(self.server.hostname, cmd,
+                                     runas=TEST_USER)
+        self.assertEqual(ret['rc'], 0)
+
+        self.momB.log_match("Job;%s;DELETE_JOB2 received" % (jid1,),
+                            max_attempts=18, interval=2)
+
+        # Verify that only 2 process left on hostB now
+        process = 0
+        self.server.pu.get_proc_info(
+            self.momB.hostname, ".*fib.*", None, regexp=True)
+        if (self.server.pu.processes is not None):
+            for key in self.server.pu.processes:
+                if ("fib" in key):
+                    process = len(self.server.pu.processes[key])
+                    print "length of the process is %d " % (process,)
+        self.assertEqual(process, 2)
+
+        # Mom logs only have message for job1 for node3
+        self.momA.log_match(
+            "Job;%s;%s.+cput=.+mem.+" % (jid1, self.hostB),
+            max_attempts=18, interval=2, regexp=True)
+
+        self.momA.log_match(
+            "Job;%s;%s.+cput=.+mem.+" % (jid2, self.hostB),
+            max_attempts=8, regexp=True,
+            existence=False)
+
+        self.momA.log_match(
+            "Job;%s;%s.+cput=.+mem.+" % (jid3, self.hostB),
+            max_attempts=5, regexp=True,
+            existence=False)
+
+        # Verify the new schedselect for job1
+        new_sched_j1 = "2:ncpus=1"
+        new_e_host_j1 = e_host_j1.replace("+%s/0" % (self.hostB,), "")
+        new_e_vnode = e_vnode.replace("+(%s:ncpus=1)" % (self.hostB,), "")
+        self.server.expect(JOB, {'job_state': "R",
+                                 "exec_host": new_e_host_j1,
+                                 "exec_vnode": new_e_vnode,
+                                 "schedselect": "1:ncpus=1+1:ncpus=1",
+                                 "Resource_List.ncpus": 2,
+                                 "Resource_List.nodect": 2}, id=jid1)
+
+        # Verify that host and vnode won't change for job2 and job3
+        self.server.expect(JOB, {'job_state': "R",
+                                 "exec_host": e_host_j2,
+                                 "exec_vnode": e_vnode,
+                                 "Resource_List.nodect": 3}, id=jid2)
+        self.server.expect(JOB, {'job_state': 'R',
+                                 "exec_host": e_host_j3,
+                                 "exec_vnode": e_vnode,
+                                 "Resource_List.nodect": 3}, id=jid3)
+
+    def test_PBS_JOBID(self):
+        """
+        Test that if jobid is not provided then it is
+        picked by env variable $PBS_JOBID
```

Comment:
I don't see you actually call pbs_release_nodes.  Also could you change the docstring to say "if -j jobid is not provided" (add the -j)

Reply:
Yes it is defined in the job script. updated the docstring